unsafe/src/main/java/org/apache/spark/unsafe/PlatformDependent.java

```diff
@@ -89,7 +89,9 @@ public static void putDouble(Object object, long offset, double value) {
     }
 
     public static long allocateMemory(long size) {
-      return _UNSAFE.allocateMemory(size);
+      long address = _UNSAFE.allocateMemory(size);
+      _UNSAFE.setMemory(address, size, (byte) 0);
```

Comment:
I don't think that we should zero out our allocated memory by default:

- In many cases, it is unnecessary and carries a performance penalty.
- `setMemory` isn't available in all versions of Java 6; this is one of the reasons why I chose not to expose it in the `UNSAFE` facade.